doc/release/upcoming_changes/15233.highlight.rst

```diff
@@ -0,0 +1,6 @@
+* As part of the continued removal of Python 2 compatibility, a sigificant
+portion of code guarded by ``PY_VERSION_HEX``, ``PY_MAJOR_VERSION``, and
+``NPY_PY3K`` was removed. This should have no effect on Python 3.5+.
+
+* A number of Py3k compatible structure have been simplified or removed.
+
```

Comment:
This is rst. The continuation lines of the bullet list must be indented to match the first line:
```
* As part ...
  this line is indented two spaces
```

But I think as a highlight, this should be something like "Code compatibility with Python versions `< 3.5` (including Python 2) was dropped from both python and C. The shims in `numpy.compat` will remain to support third-party packages, but they may be deprecated in a future release."

Then, if you want, you can add another  `compatibility` note like `15229.compatibility.rst` for this PR